diff --git a/entityreference.module b/entityreference.module
index 5f03c5b..2a3fabb 100644
--- a/entityreference.module
+++ b/entityreference.module
@@ -773,6 +773,11 @@ function entityreference_options_list($field, $instance = NULL, $entity_type = N
   $return = array();
   foreach ($options as $bundle => $entity_ids) {
     $bundle_label = check_plain($entity_info['bundles'][$bundle]['label']);
+    // We need to return unsanitized labels but getReferencableEntities()
+    // has already encoded them, so we decode them again.
+    foreach ($entity_ids as $id => $label) {
+      $entity_ids[$id] = decode_entities($label);
+    }
     $return[$bundle_label] = $entity_ids;
   }
 
